src/test/java/com/exasol/adapter/dialects/snowflake/SnowflakeVirtualSchemaIntegrationTestSetup.java

```diff
@@ -0,0 +1,252 @@
+                            .parameter("JDBC_DRIVER_PATH", JDBC_DRIVER_PATH)
+                            .mitigation("You can generate it by executing the integration test with maven.").toString(),
+                    exception);
+        }
+    }
+
+    private static void uploadVsJarToBucket(final Bucket bucket) {
+        try {
+            bucket.uploadFile(PATH_TO_VIRTUAL_SCHEMAS_JAR, VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION);
+        } catch (FileNotFoundException | BucketAccessException | TimeoutException exception) {
+            throw new IllegalStateException("Failed to upload jar to bucket " + bucket, exception);
+        }
+    }
+
+    private AdapterScript createAdapterScript(final ExasolSchema schema) {
+        final String content = "%scriptclass com.exasol.adapter.RequestDispatcher;\n" //
+                + "%jar /buckets/bfsdefault/default/" + VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION + ";\n";
+        return schema.createAdapterScript(ADAPTER_SCRIPT_EXASOL, JAVA, content);
+    }
+
+    public Statement getSnowflakeStatement() {
+        return this.snowflakeStatement;
+    }
+
+    public Statement getExasolStatement() {
+        return this.exasolStatement;
+    }
+
+    public ExasolContainer<? extends ExasolContainer<?>> getExasolContainer() {
+        return this.exasolContainer;
+    }
+
+    public VirtualSchema createVirtualSchema(final String forSnowflakeSchema,
+                                             final Map<String, String> additionalProperties) {
+        // TODO add localstack support + cleanup
+        final Map<String, String> properties = new HashMap<>(Map.of("CATALOG_NAME", databaseName, //
+                "SCHEMA_NAME", forSnowflakeSchema)); //
+        properties.putAll(additionalProperties);
+        return this.exasolFactory
+                .createVirtualSchemaBuilder("SNOWFLAKE_VIRTUAL_SCHEMA_" + (this.virtualSchemaCounter++))
+                .adapterScript(this.adapterScript).connectionDefinition(this.connectionDefinition)
+                .properties(properties).build();
+    }
+
+    public ExasolObjectFactory getExasolFactory() {
+        return this.exasolFactory;
+    }
+
+    @Override
+    public void close() {
+        try {
+            this.exasolStatement.close();
+            this.exasolConnection.close();
+            this.snowflakeStatement.close();
+            this.snowflakeConnection.close();
+            this.exasolContainer.stop();
+            // TODO add localstack support + cleanup
+            // this.snowflakeContainer.stop();
+        } catch (final SQLException exception) {
+            throw new IllegalStateException("Failed to stop test setup.", exception);
+        }
+    }
+
+    private String getTestHostIpFromInsideExasol() {
+        final Map<String, ContainerNetwork> networks = this.exasolContainer.getContainerInfo().getNetworkSettings()
+                .getNetworks();
+        if (networks.size() == 0) {
+            return null;
```

Comment:
Throw an exception here instead of returning null.

Reply:
Any suggestions? Also, there's a assertion after this function call to check it's not null.

Reply:
Ok, since there's an assert, we shouldn't bother, as discussed.